distributed/nanny.py

```diff
@@ -784,7 +779,7 @@ async def _wait_until_connected(self, uid):
                 continue
 
             if msg["uid"] != uid:  # ensure that we didn't cross queues
-                continue
+                raise RuntimeError("Encountered message from a different queue.")
```

Comment:
we're using a new queue for every process. how could these messages ever "cross"?